### ISSUE TITLE
[Bug]: Outlook Notes folder locks unread messages badge

### ISSUE THREAD

**Issue by deadmeu** [ferdium/ferdium-app#972]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-recipes/blob/develop/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-recipes/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/ferdium/ferdium-recipes/issues) for a feature request that matches the one I want to file, without success.

### Ferdium Version

6.2.0

### Recipe Name

Office 365 Outlook

### Recipe Version

1.6.0

### Steps to reproduce

1. Open Outlook
2. See Notes folder with unread message count

### Expected Behavior

Notes shouldn't be included in the unread message count

### Actual Behavior

Notes are included in the unread message count

### Screenshots

![image](https://user-images.githubusercontent.com/12111013/204438838-2ee2b950-9305-4ea2-94d3-2a81f917c2dc.png)
![image](https://user-images.githubusercontent.com/12111013/204438846-ff222974-85a8-4e93-b40d-14d4cc4b2cd2.png)

**Comment by deadmeu** [comment]
Note that you can "Empty" the Notes folder, but after a few minutes Outlook seems to add back in an empty note automatically, bringing back the badge.

**Comment by bicolorman** [comment]
Any advance in this issue? There is any workaround?

**Comment by deadmeu** [comment]
This doesn't seem to be an issue for me anymore. It seems like the permanent empty note issue no longer occurs, so the badge is not stuck on.

However, if you create a note it still triggers the badge. I'll keep this ticket open for those who use notes and don't want to be bothered by the badge.

**Comment by HigHurtenflurst** [comment]
Now that the junk mail and drafts folders now show counts of what's inside and Ferdium 6.7.1 now shows a count instead of a badge for Outlook, it now shows a highly inflated number for the badge:

![image](https://github.com/ferdium/ferdium-app/assets/10043365/b84695a5-fb91-4a2f-8fe1-09f56ef0e611)

Not sure how the counts are added together in the recipe, but maybe they can be filtered out via styling classes?

**Comment by tikkatek** [comment]
I thought I was the only one with this issue, until I decided to look at the GitHub for this project. (Mental note: always look at GitHub for OSS)

Anyhow. The code in webview.js looks llike it's iterating through all the folders and doesn't have an ignore flag of some sort.

In my use case, I am only interested in the main Inbox folder and not any of the children. So the workaround I employed was to only show favourites in the unread count.

Did this by right-clicking on the Outlook service icon, on the left-hand bar, and choosing Edit
![image](https://github.com/ferdium/ferdium-app/assets/26571381/d75ce451-6662-44b9-9afc-4e4f9316cfa3)

Then, under **Unread messasge badges**, toggle on the option for "Only show Favorites in unread count"
![image](https://github.com/ferdium/ferdium-app/assets/26571381/b9441f62-367f-4252-9303-c6adc8ea428d)

Once that is done, click on the service icon to load up your Outlook inbox, expand "Favourites" and remove anything you don't want to be counted. I removed Junk Email, Deleted Items, and Notes.
![image](https://github.com/ferdium/ferdium-app/assets/26571381/d9d10c18-4c35-49c4-a477-747a17541813)

Then you can finally right-click on the Outlook service icon again and choose "Reload" (or hit Ctrl+R on the keyboard). 

Now I only get counts of emails going directly into the Inbox folder, ignoring (per my setup) Notes,, Junk Mail, and Drafts.

Hope this helps.

**Comment by mbrozzo** [comment]
Hello @tikkatek, I had the same problem regarding inflated unread count, so I followed your advice, but I get no badge:
![Image](https://github.com/user-attachments/assets/688c09af-8f60-4734-867c-0a05671e390b)
I guess that is a bug.

Do you know of any other workaround?

Thank you in advance.

**Comment by tikkatek** [comment]
> Hello [@tikkatek](https://github.com/tikkatek), I had the same problem regarding inflated unread count, so I followed your advice, but I get no badge: ![Image](https://github.com/user-attachments/assets/688c09af-8f60-4734-867c-0a05671e390b) I guess that is a bug.
> 
> Do you know of any other workaround?
> 
> Thank you in advance.

Hey @mbrozzo. I'm afraid I haven't looked at this issue for a while as I switched from Ferdium to another free (but not FOSS) app. Please don't take that as a negative view of Ferdium as I really think it's a very capable FOSS app. My particular workflow is very demanding. 😊

**Comment by tikkatek** [comment]
Just downloaded and installed Ferdium. The Outlook service is just plain weird. It would open a browser tab for authentication, but never leaves the browser to log into the app. I then changed the service to self-hosted and used the URL [https://outlook.office365.com/mail/](https://outlook.office365.com/mail/) as a workaround. But every time the service reloads it loses the login and asks for the login again, but does skip the password.

Aaaaanyway. You're right. My workaround to use Favourites no longer works and there's no badge. But if I disable the Favourites, I do get a badge, but the recycle bin and junk still gets counted. 😔

**Comment by mbrozzo** [comment]
Thank you anyways @tikkatek, I appreciate you taking time to test and answer

**Comment by MrDanielHarka** [comment]
I'm also having issues with "Only show Favorites in unread count".

If I turn it on and I set only my inbox as favorite, then it's not showing the unread icon.
If I turn it off then I get a high unread message count, because I delete many emails without reading them.

It would be really nice to have this working.

Edit:
I have also set a self-hosted url. The above happens for both of the below urls.
- https://outlook.office.com/mail/
- https://outlook.office365.com/mail/

**Comment by TimBurris** [comment]
i have same issue as @MrDanielHarka  the "Only show Favorites in unread count" does not appear to work. 
the fix was to edit the recipe:
C:\Users\[user]\AppData\Roaming\Ferdium\recipes\office365-owa\webview.js

change line 75 to:
? collectCounts('div[role="group"][aria-labelledby="favoritesRoot"]')

essentially OWA source has changed, the previous selector is simply inaccurate. 
who knows how long this fix will last
